src/components/tl/ucp/alltoallv/alltoallv_onesided.c

```diff
@@ -0,0 +1,113 @@
+    size_t           rdt_size = ucc_dt_size(TASK_ARGS(task).dst.info_v.datatype);
+    ucc_rank_t       peer;
+    size_t           sd_disp, dd_disp, data_size;
+
+    ucc_tl_ucp_task_reset(task, UCC_INPROGRESS);
+
+    /* perform puts to each member i using that displacement */
+    for (peer = (grank + 1) % gsize; task->onesided.put_posted < gsize;
+         peer = (peer + 1) % gsize) {
+        sd_disp =
+            ucc_coll_args_get_displacement(&TASK_ARGS(task), s_disp, peer) *
+            sdt_size;
+        dd_disp =
+            ucc_coll_args_get_displacement(&TASK_ARGS(task), d_disp, peer) *
+            rdt_size;
+        data_size =
+            ucc_coll_args_get_count(&TASK_ARGS(task),
+                                    TASK_ARGS(task).src.info_v.counts, peer) *
+            sdt_size;
+
+        UCPCHECK_GOTO(ucc_tl_ucp_put_nb((void *)PTR_OFFSET(src, sd_disp),
+                                        (void *)PTR_OFFSET(dest, dd_disp),
+                                        data_size, peer, team, task),
+                      task, out);
+        UCPCHECK_GOTO(ucc_tl_ucp_atomic_inc(pSync, peer, team), task, out);
+    }
+    return ucc_progress_queue_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+out:
+    return task->super.status;
+}
+
+void ucc_tl_ucp_alltoallv_onesided_progress(ucc_coll_task_t *ctask)
+{
+    ucc_tl_ucp_task_t *task  = ucc_derived_of(ctask, ucc_tl_ucp_task_t);
+    ucc_tl_ucp_team_t *team  = TASK_TEAM(task);
+    ucc_rank_t         gsize = UCC_TL_TEAM_SIZE(team);
+    long              *pSync = TASK_ARGS(task).global_work_buffer;
+
+    if ((*pSync < gsize) ||
+        (task->onesided.put_completed < task->onesided.put_posted)) {
+        ucp_worker_progress(UCC_TL_UCP_TEAM_CTX(team)->worker.ucp_worker);
+        return;
+    }
+
+    pSync[0]           = 0;
+    task->super.status = UCC_OK;
+}
+
+ucc_status_t ucc_tl_ucp_alltoallv_onesided_init(ucc_base_coll_args_t *coll_args,
+                                                ucc_base_team_t      *team,
+                                                ucc_coll_task_t     **task_h)
+{
+    ucc_tl_ucp_team_t *tl_team = ucc_derived_of(team, ucc_tl_ucp_team_t);
+    ucc_tl_ucp_task_t *task;
+    ucc_status_t       status;
+
+    ALLTOALLV_TASK_CHECK(coll_args->args, tl_team);
+    if (!(coll_args->args.mask & UCC_COLL_ARGS_FIELD_GLOBAL_WORK_BUFFER)) {
+        tl_error(UCC_TL_TEAM_LIB(tl_team),
+                 "global work buffer not provided nor associated with team");
+        status = UCC_ERR_NOT_SUPPORTED;
+        goto out;
+    }
+    if (coll_args->args.mask & UCC_COLL_ARGS_FIELD_FLAGS) {
+        if (!(coll_args->args.flags & UCC_COLL_ARGS_FLAG_MEM_MAPPED_BUFFERS)) {
+            tl_error(UCC_TL_TEAM_LIB(tl_team),
+                     "non memory mapped buffers are not supported");
+            status = UCC_ERR_NOT_SUPPORTED;
+            goto out;
+        }
+    }
+    if (!(coll_args->args.flags & UCC_COLL_ARGS_FLAG_ONESIDED_VECTOR)) {
```

Comment:
I might be missing something but from what I understand, this flag is necessary for alltoallv_onesided, and is not relevant in any other context. I wonder why this is needed in the API.
Could you explain why we need this flag and how the user can interact with it ?
Should we add checks in the other collective that the flag is not set ?